mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -124,9 +130,35 @@ def open_multifile_dataset(fileNames, calendar, simulationStartTime=None,
                                  startDate=startDate,
                                  endDate=endDate)
 
-    ds = xarray.open_mfdataset(fileNames,
-                               preprocess=preprocess_partial,
-                               decode_times=False, concat_dim='Time', **kwargs)
+    # we don't want to decode times, since we're typically using dates
+    # outside the allowed range
+    kwargs['decode_times'] = False
+    # files differ along the Time dimension
+    kwargs['concat_dim'] = 'Time'
+
+    autoclose = ((autocloseFileCount is not None) and
+                 len(fileNames) > autocloseFileCount)
+
+    openedWithAutoclose = False
+    if autoclose:
+        try:
+            ds = xarray.open_mfdataset(fileNames,
+                                       preprocess=preprocess_partial,
+                                       autoclose=True, **kwargs)
+            openedWithAutoclose = True
+        except TypeError as e:
+            if 'autoclose' in str(e):
+                # This indicates that xarray version doesn't support autoclose
```

Comment:
Check if `autoclose` is somewhere in the error message (so we don't print an unhelpful warning when other arguments besides `autoclose` are the problem).